### PR TITLE
Test Setup

### PR DESCRIPTION
**Pull request by mboneil10**
Part 1 of 3 for #21

### CODE REVIEW

File: config/initializers/devise.rb [hunk 1]
```diff
@@ -1,265 +1,10 @@
...
-  # find_for_authentication method and considered in your model lookup. For instance,
-  # if you set :request_keys to [:subdomain], :subdomain will be used on authentication.
-  # The same considerations mentioned for authentication_keys also apply to request_keys.
-  # config.request_keys = []
-
-  # Configure which authentication keys should be case-insensitive.
-  # These keys will be downcased upon creating or modifying a user and when used
-  # to authenticate or find a user. Default is :email.
+  
+  config.mailer_sender = 'stop-project@pvtaapps.com'
```
[the remark this answers is not in the document]
Comment (mboneil10) [reply config/initializers/devise.rb]: Oh. Oops. No, we don't use a mail sender in this app.

File: spec/rails_helper.rb [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+#
+# Dir[Rails.root.join('spec/support/**/*.rb')].each { |f| require f }
+
+# Checks for pending migrations and applies them before tests are run.
+# If you are not using ActiveRecord, you can remove this line.
+ActiveRecord::Migration.maintain_test_schema!
+
+RSpec.configure do |config|
+  # Remove this line if you're not using ActiveRecord or ActiveRecord fixtures
+  config.fixture_path = "#{::Rails.root}/spec/fixtures"
```
[the remark this answers is not in the document]
Comment (mboneil10) [reply spec/rails_helper.rb]: Ok.

File: spec/spec_helper.rb [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+RSpec.configure do |config|
+  config.order = :random
+  config.include FactoryBot::Syntax::Methods
+  config.include Devise::Test::ControllerHelpers, type: :controller
+  config.include Devise::Test::IntegrationHelpers, type: :system
+  config.before :all do
+    FactoryBot.reload
+  end
+  config.expect_with :rspec do |expectations|
+    expectations.include_chain_clauses_in_custom_matcher_descriptions = true
```
[the remark this answers is not in the document]
Comment (mboneil10) [reply spec/spec_helper.rb]: No idea, good catch. I'll look over this whole file again.

File: spec/spec_helper.rb [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+  end
+  config.mock_with :rspec do |mocks|
+    mocks.verify_partial_doubles = true
```
[the remark this answers is not in the document]
Comment (mboneil10) [reply spec/spec_helper.rb]: Yeah...

File: spec/rails_helper.rb [hunk 5]
```diff
@@ -0,0 +1,57 @@
+# This file is copied to spec/ when you run 'rails generate rspec:install'
+require 'spec_helper'
```
[the remark this answers is not in the document]
Comment (mboneil10) [reply spec/rails_helper.rb]: Thanks @dfaulken. I'll get rid of one.

File: config/initializers/devise.rb [hunk 6]
```diff
@@ -1,265 +1,10 @@
...
   config.case_insensitive_keys = [:email]
-
-  # Configure which authentication keys should have whitespace stripped.
-  # These keys will have whitespace before and after removed upon creating or
-  # modifying a user and when used to authenticate or find a user. Default is :email.
+  config.secret_key = '70c47a4a4303439f42c3064ee46b19f26cd584ddec55047c45f289a2e7d6855aa3dbce0dcc6fdf84a0595dee2a79784050a7f5692195ad9e83ec966088b3fa3f'
```
[the remark this answers is not in the document]
Comment (mboneil10) [reply config/initializers/devise.rb]: Whoops.

File: spec/spec_helper.rb (hunk 4 above)
[the remark this answers is not in the document]
Comment (mboneil10) [reply spec/spec_helper.rb]: [You should set this unless you have a good reason not to](https://relishapp.com/rspec/rspec-mocks/docs/verifying-doubles/partial-doubles)?

File: .gitignore [hunk 7]
```diff
@@ -13,3 +13,8 @@
 
 /node_modules
 /yarn-error.log
+
+# Ignore all of coverage except for last_run.json
```
Comment (Anbranin) [inline .gitignore]: Why?

File: Gemfile [hunk 8]
```diff
@@ -23,6 +23,17 @@ group :development do
...
-  gem 'pry-byebug'
   gem 'timecop', '~> 0.9.1'
 end
+
+group :development, :test do
+  gem 'factory_bot_rails'
+  gem 'pry-byebug'
+end
+
+group :test do
```
Comment (Anbranin) [inline Gemfile]: You're going to need the gem `puma`. See [here](https://github.com/umts/screaming-dinosaur/pull/106)

File: .gitignore (hunk 7 above)
Comment (dfaulken) [reply .gitignore]: It's what we do on all of our other projects.

File: Gemfile (hunk 8 above)
Comment (mboneil10) [reply Gemfile]: Hooray! thanks

File: .gitignore (hunk 7 above)
Comment (Anbranin) [reply .gitignore]: oh

**Pull request merged**